contracts/Controller.sol

```diff
@@ -379,12 +461,36 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
      * @return amount of collateral that can be taken out
      */
     function getProceed(address _owner, uint256 _vaultId) external view returns (uint256) {
-        MarginVault.Vault memory vault = getVault(_owner, _vaultId);
+        (MarginVault.Vault memory vault, uint256 typeVault, ) = getVault(_owner, _vaultId);
 
-        (uint256 netValue, ) = calculator.getExcessCollateral(vault);
+        (uint256 netValue, ) = calculator.getExcessCollateral(vault, typeVault);
         return netValue;
     }
 
+    /**
+     * @notice check if a vault is liquidatable in a specific round id
+     * @param _owner vault owner address
+     * @param _vaultId vault id to check
+     * @param _roundId chainlink round id to check vault status at
+     * @return isUnderCollat, true if vault is undercollateralized, the price of 1 repaid otoken and the otoken collateral dust amount
+     */
+    function isLiquidatable(
+        address _owner,
+        uint256 _vaultId,
+        uint256 _roundId
+    )
+        external
+        view
+        returns (
+            bool,
+            uint256,
+            uint256
+        )
+    {
+        (, bool isUnderCollat, uint256 price, uint256 dust) = _isLiquidatable(_owner, _vaultId, _roundId);
+        return (isUnderCollat, price, dust);
```

Comment:
Here I used `isUnderCollat` instead of `isLiquidatable` to avoid the compiler warning re shadowing vars